UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/ThirdParty/JSBSim.Build.cs

```diff
@@ -5,45 +5,88 @@
 
 public class JSBSim : ModuleRules
 {
-	public JSBSim(ReadOnlyTargetRules Target) : base(Target)
-	{
-		Type = ModuleType.External;
-		
-		// This folder is where the JSBSimForUnreal build has been generated
-		string JSBSimLocalFolder = "JSBSim";
-
-		if (Target.Platform == UnrealTargetPlatform.Win64)
-		{
-			string LibFolderName = "Lib";
-
-			// When working in debug mode, try to use the Debug version of JSBSim
-			if (Target.Configuration == UnrealTargetConfiguration.Debug)
-			{
-				// Source\ThirdParty\JSBSim\LibDebug
-				string DebugLibsPath = Path.Combine(ModuleDirectory, JSBSimLocalFolder, LibFolderName + "Debug");
-				if (Directory.Exists(DebugLibsPath))
-				{
-					System.Console.WriteLine(string.Format("Found Debug libraries for JSBSim in {0}", DebugLibsPath));
-					LibFolderName += "Debug";
-				}
-			}
-
-			// Include headers
-			PublicSystemIncludePaths.Add(Path.Combine(ModuleDirectory, JSBSimLocalFolder, "Include"));
-
-			// Link Lib
-			string LibPath = Path.Combine(ModuleDirectory, JSBSimLocalFolder, LibFolderName);
-			PublicAdditionalLibraries.Add(Path.Combine(LibPath, "JSBSim.lib"));
-			
-			// Stage DLL along the binaries files
-			string DllFullPath = Path.Combine(LibPath, "JSBSim.dll");
-			if (!File.Exists(DllFullPath))
+    public JSBSim(ReadOnlyTargetRules Target) : base(Target)
+    {
+        Type = ModuleType.External;
+
+        bool bJSBSimSupported = Target.Platform == UnrealTargetPlatform.Win64 ||
+                                Target.Platform == UnrealTargetPlatform.Mac || 
+                                Target.Platform == UnrealTargetPlatform.Linux || 
+                                Target.Platform == UnrealTargetPlatform.Android;
+
+        if (!bJSBSimSupported) return;
+
+        if (Target.Platform == UnrealTargetPlatform.Win64)
+            SetupWindowsPlatform();
+        else
+            SetupUnixPlatform();
+    }
+
+    private void SetupWindowsPlatform()
+    {
+        string JSBSimLocalFolder = "JSBSim";
+        string LibFolderName = "Lib";
+
+        // When working in debug mode, try to use the Debug version of JSBSim
+        if (Target.Configuration == UnrealTargetConfiguration.Debug)
```

Comment:
Following the PR #912, this line must be updated:
```suggestion
        if (Target.Configuration == UnrealTargetConfiguration.DebugGame)
```